### PR TITLE
NRP-3606-Added endpoint for query wizard post

### PR DESCRIPTION
**Pull request by lgrzybowski**
I added post method for new query wizard.

### CODE REVIEW

**Comment by lgrzybowski** [comment]
@jonathanchrisp @benSlaughter @leemcgeever 
I changed travis setting as new version of tins package (1.7.0) requires Ruby in version >= 2.0 
https://github.com/flori/tins

File: lib/bwapi/client/brandwatch/usage_report.rb [hunk 1]
```diff
@@ -5,13 +5,13 @@ module Brandwatch
       module UsageReport
         # Get the Clients Mention Usage report
         #
-        # @param clientId [Integer] Id of client
-        # @param startDate [DateTime] Report startDate
-        # @param endDate [DateTime] Report endDate
+        # @param client_id [Integer] Id of client
```
Comment (jonathanchrisp) [inline lib/bwapi/client/brandwatch/usage_report.rb]: Good spot :+1:

File: lib/bwapi/client/projects/queries.rb [hunk 2]
```diff
@@ -61,6 +61,32 @@ def create_query(project_id, opts = {})
...
+        # @option opts [Array] languages Query languages
+        # @option opts [String] twitterScreenName Tracked twitter screen name
+        # @option opts [String] industry Industry of the query
+        # @option opts [String] usecase The Usecase the Query was created for
+        # @option opts [String] type Type of query
+        # @option opts [Object] queryWizardConfiguration JSON object containing configuration for Query Wizard
+        # @option opts [Date] creationDate Date the query was created on
+        # @option opts [Date] lastModificationDate Modification date of the query
+        # @return [Hash] New query wizard
+        def create_query_wizard(project_id, opts = {})
```
Comment (jonathanchrisp) [inline lib/bwapi/client/projects/queries.rb]: could we rename this to `create_wizard_query`, is this going to replace the standard query?

File: lib/bwapi/client/projects/queries.rb [hunk 3]
```diff
@@ -61,6 +61,32 @@ def create_query(project_id, opts = {})
           post "projects/#{project_id}/queries", opts
         end
 
+        # Create a wizard query in project
+        #
+        # @param project_id [Integer] Id of project
+        # @param opts [Hash] options hash of parameters
+        # @option opts [String] id Id of the query
```
Comment (jonathanchrisp) [inline lib/bwapi/client/projects/queries.rb]: We should rename `opts` to the appropriate parameter, please do this for the other instances below.

File: lib/bwapi/client/projects/queries.rb (hunk 2 above)
Comment (lgrzybowski) [inline lib/bwapi/client/projects/queries.rb]: No we will have two entry point for building query, easy one and for more advanced users (current query builder).

Comment (jonathanchrisp) [inline lib/bwapi/client/projects/queries.rb]: Sounds good, yeah, I think `create_wizard_query` is better IMHO.

**Comment by jonathanchrisp** [comment]
@lgrzybowski - out of interest what do we use the `tins` package for?

**Comment by lgrzybowski** [comment]
@jonathanchrisp I changed the name and added two more endpoint as they are done now. I hope it can be merge as part of that PR. 
Regarding your question as I see tins is used by rubocop
```
rubocop (0.35.1)
      ....
      tins (<= 1.6.0)
```

File: lib/bwapi/client/projects/queries.rb [hunk 4]
```diff
@@ -61,6 +70,32 @@ def create_query(project_id, opts = {})
           post "projects/#{project_id}/queries", opts
         end
 
+        # Create a wizard query in project
+        #
+        # @param project_id [Integer] Id of project
+        # @param opts [Hash] options hash of parameters
+        # @option opts [Integer] id Id of the query
```
Comment (leemcgeever) [inline lib/bwapi/client/projects/queries.rb]: do we need `id` and `Id`?

File: lib/bwapi/client/projects/queries.rb [hunk 5]
```diff
@@ -42,7 +51,7 @@ def get_query(project_id, query_id)
         #
         # @param project_id [Integer] Id of project
         # @param opts [Hash] options hash of parameters
-        # @option opts [String] id Id of the query
+        # @option opts [Integer] id Id of the query
         # @option opts [String] name Name of the project
```
Comment (leemcgeever) [inline lib/bwapi/client/projects/queries.rb]: do we need `id` and `Id`?

File: lib/bwapi/client/projects/queries.rb [hunk 6]
```diff
@@ -85,6 +120,32 @@ def update_query(project_id, query_id, opts = {})
           put "projects/#{project_id}/queries/#{query_id}", opts
         end
 
+        # Update an existing wizard query in project
+        #
+        # @param project_id [Integer] Id of project
+        # @param opts [Hash] options hash of parameters
+        # @option opts [Integer] id Id of the query
```
Comment (leemcgeever) [inline lib/bwapi/client/projects/queries.rb]: do we need `id` and `Id`?

File: lib/bwapi/client/projects/queries.rb [hunk 7]
```diff
@@ -85,6 +120,32 @@ def update_query(project_id, query_id, opts = {})
...
+        # @option opts [String] type Query type
+        # @option opts [Array] includedTerms Included terms of terms query
+        # @option opts [Array] contextTerms Content terms of the query
+        # @option opts [Array] excludedTerms Excluded terms of thequery
+        # @option opts [Array] languages Query languages
+        # @option opts [String] twitterScreenName Tracked twitter screen name
+        # @option opts [String] industry Industry of the query
+        # @option opts [String] usecase The Usecase the Query was created for
+        # @option opts [String] type Type of query
+        # @option opts [Object] queryWizardConfiguration JSON object containing configuration for Query Wizard
```
Comment (leemcgeever) [inline lib/bwapi/client/projects/queries.rb]: again  do we need type twice?

**Comment by leemcgeever** [comment]
LGTM :+1:

**Comment by jonathanchrisp** [comment]
LGTM - merging :+1:

**Pull request merged**